code/go/internal/pkgpath/files.go

```diff
@@ -66,20 +66,70 @@ func (f File) Values(path string) (interface{}, error) {
 		return nil, errors.Wrap(err, "reading file content failed")
 	}
 
-	var v interface{}
+	v := make(map[string]interface{})
 	if fileExt == "yaml" || fileExt == "yml" {
-		if err := yaml.Unmarshal(contents, &v); err != nil {
-			return nil, errors.Wrapf(err, "unmarshalling YAML file failed (path: %s)", f.fsys.Path(fileName))
+		config, err := yaml.NewConfig(contents)
+		if err != nil {
+			return nil, errors.Wrapf(err, "parsing yaml file failed (path: %s)", f.fsys.Path(fileName))
 		}
+
+		config.Unpack(&v)
```

Comment:
missing check the error returned by the Unpack function

```suggestion
		if err = config.Unpack(&v); err != nil {
			return nil, errors.Wrap(err, "can't unpack file (path: %s)", f.fsys.Path(fileName))
		}
```